utils/frame/remote-externalities/src/lib.rs

```diff
@@ -609,7 +608,14 @@ where
 
 		let mut sp = Spinner::with_timer(Spinners::Dots, "Inserting keys into DB...".into());
 		let start = Instant::now();
-		pending_ext.batch_insert(key_values.clone().into_iter().map(|(k, v)| (k.0, v.0)));
+		pending_ext.batch_insert(key_values.clone().into_iter().filter_map(|(k, v)| {
+			// Don't insert the child keys here, they need to be inserted seperately with all their
+			// data in the load_child_remote function.
+			match is_default_child_storage_key(&k.0) {
+				true => None,
+				false => Some((k.0, v.0)),
+			}
+		}));
```

Comment:
I think we want to have the result of load_top_remote to contain the child trie root (load_child_remote seems to use this results).

In my branch I did export (with no filter) to a file and then apply filter on load, but did not cover the case where we do not load from the exported file.
https://github.com/cheme/substrate/blob/e27aedbedefe145d4c9a4db6fc846ea1357bd18c/utils/frame/remote-externalities/src/lib.rs#L933 but this only do the trick when loading from a snapshot/file, basically this branch is a hack to be able to run lot of tests.

So probably here we want pending_ext to not contains the child trie roots as you did, but also modify rpc_get_pairs_paged to return the different child trie root (the filtered content), and pass them explicitely to load_child_remote (have load child remote get child root from a param and not from top_kv).

Ideally it would be better to have all child trie root always in top_kv, as it is the real state, and fix the related issue (I don't remember well what it is, but I indeed had to filter in my branch too).

Reply:
Oh yes, I misread, sorry.